ehrapy/anndata/_feature_specifications.py

```diff
@@ -65,17 +129,41 @@ def infer_feature_types(adata: AnnData, layer: str | None = None, output: Litera
 def check_feature_types(func):
     @wraps(func)
     def wrapper(adata, *args, **kwargs):
+        # Account for class methods that pass self as first argument
+        _self = None
+        if not isinstance(adata, AnnData) and len(args) > 0 and isinstance(args[0], AnnData):
+            _self = adata
+            adata = args[0]
+            args = args[1:]
+
         if FEATURE_TYPE_KEY not in adata.var.keys():
-            raise ValueError("Feature types are not specified in adata.var. Please run `infer_feature_types` first.")
-        np.all(adata.var[FEATURE_TYPE_KEY].isin([CATEGORICAL_TAG, CONTINUOUS_TAG, DATE_TAG]))
+            infer_feature_types(adata, output=None)
+            logger.warning(
+                f"Feature types were inferred and stored in adata.var[{FEATURE_TYPE_KEY}]. Please verify and adjust if necessary using `ep.ad.correct_feature_types`."
```

Comment:
Do we need to tell people how to verify?

Reply:
I think it can't hurt, so I added a reference to `ep.ad.feature_type_overview`.